internals/api/service.go

```diff
@@ -23,13 +23,32 @@ var (
 	)
 )
 
+type ServiceType string
+
+// Service types
+const (
+	ServiceTypeAWS ServiceType = "aws"
+	ServiceTypeRSA ServiceType = "rsa"
+)
+
+type ServiceMetaDataKey string
+
+// Service metadata keys
+const (
+	ServiceMetaDataAWSKMSKey ServiceMetaDataKey = "kms_key"
+	ServiceMetaDataAWSRole   ServiceMetaDataKey = "role"
```

Comment:
Should this be `role`, `role_arn`, `arn` or `aws_role`?

Reply:
I think `role` suffices.
ARN is the format in which the role is given. We typically don't describe the format in the name.
We can name it `aws_role`. Might be a bit redundant, as the type is already `aws`. I'm fine with both.

Reply:
I think it's best to distinguish between type specific fields. `aws_role` or `role_arn` implies that it's on AWS. Yes the type is already of `aws`, but I can imagine other types also using the `role` key. Then I'd rather be explicit. Small thing though.

Reply:
Oke, let's make it `aws_role` and `aws_kms_key` then :+1: